### ISSUE TITLE
Keys swapped

### ISSUE THREAD

**Issue by uberrice** [JellyTitan/Sofle-Pico#4]
It seems like there is something not quite right with the config.

I am writing from the Sofle Pico right now, so, working it is!

However, at least for my board, it looks like two keys are swapped. Specifically the two keys on the right thumbcluster, left side - the big one and the one next to it. SW26 and 27. Whatever button press is configured for SW26 happens on SW27 and vice versa. Only happens on the right side. Any idea why? Do you have the same behaviour?

**Comment by uberrice** [comment]
![image](https://github.com/JellyTitan/Sofle-Pico/assets/15884708/2998c0b1-3155-43b8-8d83-88e2c360e216)
![image](https://github.com/JellyTitan/Sofle-Pico/assets/15884708/b5cb0e83-b752-4b7a-8704-339acdcd811d)
The keymap looks fine as far as I can tell. The issue seems to be somewhere in the config.

**Comment by JellyTitan** [comment]
I found and fixed a switched matrix config:
![image](https://github.com/JellyTitan/Sofle-Pico/assets/72168556/fa987f70-cd7c-4a52-893a-f7d638412b05)

I just pushed up the fix to my QMK sofle_pico. 

I haven't updated the Uf2 files in this repo though - so the drag and drop config is still outdated. If you want to try that method of flashing - i'll update them now.

Since your OLED's are soldered - do you still have access to the `BOOT` buttons on the PCB? Or are you entering boot mode some other way? (Asking because i pulled out `BOOTMAGIC` during cleanup, and i haven't added it back in yet).

**Comment by JellyTitan** [comment]
Your setups looking great so far! 
Was the OLED issue with the socket it was plugged into? I'm a little worried that there is no relationship between the corner circles and the pin sequence . . .
I cribbed that qmk logo from another board - i think it's formatted for a 128x32 OLED. I'll paw through the repo and see if somebody else made one of the appropriate size. 
I'm having intermittent issues with `B` key registering keypresses. It's not the switch, and i don't think it's the hotswop
=

**Comment by uberrice** [comment]
The OLED issue was simply me not understanding you actually need to add it to your config ;) the 'noise screen' is the startup sequence, the screen being updated chunk by chunk.

I noticed that if right is the usb-connected keyboard, the menu is upside down.

Also, the qmk logo periodically appears on my right screen - is that on purpose?

**Comment by uberrice** [comment]
As for intermittent issues - I also had that problem, in my case simply touching up the soldering somewhat fixed it (those solder joints were ugly!)

Oh. Also - no need to set a max RGB value - I changed it back to 255 and my board is not even getting warm with all of them on full blast for hours.

**Comment by JellyTitan** [comment]
For the intermittent QMK logo - the root cause it the startup/sleep behavior. I haven't quite figured out how to resolve that. 
The screens used to timeout after a few minutes, then turn off. Now when they timeout, I'm seeing the startup screen behavior instead. I'll dig through the OLED API to figure out the correct way to handle that. I think OLED's are the last big dev chunk. (I'd like to include Bongo cat at 128x64, as well as improve the UI legability for layer state).

Thanks for checking on the max brightness of the LED's - I"ll crank that setting to max in the config, but leave it in there. (I expect it may be needed if the level shifter is bypassed).

**Comment by LoopControl** [comment]
I have nothing to contribute but just read through the three issues you 2 have been discussing this on and it's great to see the collaboration.

I'm a relative newbie to custom keyboards having only built a macropad so far but I hope to build this Sofle-Pico keyboard soon. Would it be relatively OK for me to order the current PCBs as is or should I continue waiting for more testing/revisions?

**Comment by uberrice** [comment]
Hi @LoopControl. I ordered my own revision (that got merged in at some point I think) - that definitely worked for me without any hw patches - the voltage shifter components are not needed, so you need to bridge the solder jumper on the board. I'd recommend using 6/7mm spacers or something. I used 8mm I think, which causes the board itself to be 'suspended' between the 3d printed plates. Other option would be to make the bottom plate thicker, with slots for the spacers to fit into down to the 'original height'. Once the switches are installed, it's no real issue anymore, however. 

I don't know what @JellyTitan changed since then other than the LED cutouts, he'll have to comment on that.

**Comment by JellyTitan** [comment]
@LoopControl I'm delighted that you're interested! While the [v3.5 release](https://github.com/JellyTitan/Sofle-Pico/releases/tag/v3.5) is working - I'd recommend waiting another week. 
There's some changes I'd like to validate. Everything after [3.5.2 ](https://github.com/JellyTitan/Sofle-Pico/blob/main/build_log.md#v352-12-15-23) has not been tested yet. (v3.5.4 came in yesterday, I'm hoping to put them together this weekend).
I refactored the LED footprint again, added a 'handedness pin' as well as tweaked the power circuit based on what I learned about routing from @ubberices' contrib, and I pulled the vbus/vsys diode after checking the board variants with a thermal imager. There's a  hotspot on that diode. I cribbed that part of the circuit from the Junco, and I think it can probably be removed, but I won't know until i test the board. I did open an [issue with Junco](https://github.com/daneski13/Junco/issues/3) to see if there's some data that I'm missing, or if there's any additional insight to be had.

**Comment by LoopControl** [comment]
Thanks for the responses. I'll wait for the newer revision before ordering the PCB.

I've been looking at keycaps/switches and am thinking of these "low-profile" keys from Nuphy. Would these be compatible with this PCB?
* Switches - https://nuphy.com/collections/switches?filter.p.m.custom.compatibility=Low-Profile+Keys
* Keycaps - https://nuphy.com/collections/keycaps?filter.p.m.custom.compatibility_=Low-Profile+Keys

**Comment by JellyTitan** [comment]
@LoopControl 
The first link are for "Gateron low profile" switches which have a different [footprint](https://github.com/siderakb/key-switches.pretty). 
The Sofle Pico currently uses MX footprints, so those won't work. 
However - it's not a heavy lift to swap out footprints.  Probably a few hours worth of work to swap em in, then update the traces.
Right now the two 'hot' footprint styes are MX and Choc. Boards that support other styles are rare because the demand is low.
Last year KiCad 7 was released, and it made it easier to make custom footprints. I tried to make a footprint for the [Cantor that accommodates 7 different types of switches](https://github.com/JellyTitan/Cantor-Master-Key), including MX, Choc, and Gateron low profile. The pins on the footprints are all just a little different, so it comes out looking like swiss cheese, and really not working well at all. Ultimately i found that  without three solid points of contact, the switches would be 'jiggly'. The jiggle would require keyplates, and the keyplates added cost & complexity. I abandoned the project.

![image](https://github.com/JellyTitan/Sofle-Pico/assets/72168556/7c89c4dd-bd19-4d7c-85ed-e41d4deabbca)

If you feel strongly about the gateron low profile footprint, I'd recommend waiting until the Sofle Pico is released and then forking the branch and adding the footprints yourself, or opening an issue with a feature request label so that it can be worked on someday.
